From be52e505a7f189a68ddb73c4583d38c0fd60f6d3 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Fri, 13 Apr 2018 09:41:58 -0500
Subject: [PATCH] Use existing GitInfo* if provided

This allows the caller to avoid the git calls when not building in a repository.
---
 build/GitCommitInfo.targets | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/build/GitCommitInfo.targets b/build/GitCommitInfo.targets
index 6ac729e50..baf01a253 100644
--- a/build/GitCommitInfo.targets
+++ b/build/GitCommitInfo.targets
@@ -1,12 +1,14 @@
 <Project ToolsVersion="15.0">
   <Target Name="WriteGitCommitInfoProps">
     <Exec Command="git rev-list --count HEAD" 
-          ConsoleToMSBuild="true">
+          ConsoleToMSBuild="true"
+          Condition=" '$(GitInfoCommitCount)' == '' ">
       <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitCount" />
     </Exec>
 
     <Exec Command="git rev-parse HEAD" 
-          ConsoleToMSBuild="true">
+          ConsoleToMSBuild="true"
+          Condition=" '$(GitInfoCommitHash)' == '' ">
       <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitHash" />
     </Exec>
 
-- 
2.16.1.windows.4

